articles/api-auth/dynamic-client-registration.md

```diff
@@ -0,0 +1,205 @@
+---
+description: How to dynamically register Clients with Auth0, using the Management APIv2.
+toc: true
+---
+
+# Dynamic Client Registration
+
+Dynamic Client Registration is a feature, based on the [OpenID Connect Dynamic Client Registration specification](https://openid.net/specs/openid-connect-registration-1_0.html), that enables you to to register clients dynamically, using the [Management APIv2](/api/management/v2). This way you can allow third party developers to register and use clients.
+
+All clients, registered dynamically with Auth0, have the following characteristics:
+
+- The [ID tokens](/tokens/id-token) generated for these clients, hold minimum user profile information. In order to retrieve the full user profile, you need to get an access token and use the [/userinfo endpoint](/api/authentication#get-user-info).
+
+- These clients cannot use [ID tokens](/tokens/id-token) to invoke [Management APIv2](/api/management/v2) endpoints. Instead, they should [get a Management APIv2 Token](/api/management/v2/tokens#how-to-get-a-management-apiv2-token). Note that the client should be granted the `current_user_*` scopes, as required by each endpoint.
```

Comment:
The linked document shows how to obtained such scope thru a client credentials exchange. That exchange won't work because that's authorization for clients; the notion of user is not present there.

To get these scopes, clients will need to make a call to `/authorize` with the following,
- `audience=API2_URL`
- `scopes=read:current_user update:current_user_metadata`

This will give the client an access token to API2 with the selected scopes for that particular user.